### PR TITLE
Support for loading range-compressed data

### PR DESCRIPTION
**Pull request by nrother**
We have a lot of pre-existing data, often already range-compressed. This PR adds code to support loading such captures and process them.

I also added a somewhat formal spec of the format the simulator can load, called "sardata".

### CODE REVIEW

File: sarsim/sardata.py [hunk 1]
```diff
@@ -83,8 +118,8 @@ def _load_cfg_values(self):
         sim.image_stop_x = image_region[2]
         sim.image_stop_y = image_region[3]
 
-        sim.image_count_x = cfg['params'].getint('gbp_image_pixels_range', fallback=1024)
-        sim.image_count_y = round(sim.image_count_x *
+        sim.image_count_y = cfg['params'].getint('gbp_image_pixels_ range', fallback=1024)
```
Comment (cbcf) [inline sarsim/sardata.py:121]: Is this correct?

File: sarsim/sardata.py [hunk 2]
```diff
@@ -13,50 +15,83 @@ class SarData(object):
...
+            sim.fmcw_stop_frequency = cfg['params'].getfloat('center_frequency') + cfg['params'].getfloat('bandwidth')
+            sim.fmcw_ramp_duration = 0
 
         sim.azimuth_start_position = cfg['params'].getfloat('azimuth_start_position')
         sim.azimuth_stop_position = cfg['params'].getfloat('azimuth_end_position')
         # Note: azimuth_speed_limit not used YET - TODO: Add when we introduce velocity
         sim.azimuth_count = cfg['params'].getint('azimuth_count')
 
-        sim.azimuth_compression_beam_limit = cfg['params'].getfloat('gbp_beam_limit', fallback=30)
+        sim.azimuth_compression_beam_limit = cfg['params'].getfloat('gbp_beam_limit', fallback=30)
```
Comment (cbcf) [inline sarsim/sardata.py:83]: Exra Whitespace

File: sarsim/sardata.py (hunk 1 above)
Comment (cbcf) [reply sarsim/sardata.py:121]: xy swap? also "pixels_ range"

**Review by cbcf — CHANGES_REQUESTED** [review]

File: sarsim/sardata.py (hunk 2 above)
Comment (nrother) [reply sarsim/sardata.py]: Fixed

File: sarsim/sardata.py (hunk 1 above)
Comment (nrother) [reply sarsim/sardata.py]: Space is fixed, X/Y should be correct (Y is Range)

**Review by cbcf — APPROVED** [review]

**Pull request merged**